src/NuGet.Services.AzureSearch/SearchService/SearchResponseBuilder.cs

```diff
@@ -110,6 +111,50 @@ public class SearchResponseBuilder : ISearchResponseBuilder
             };
         }
 
+        public AutocompleteResponse AutocompleteFromSearch(
+            AutocompleteRequest request,
+            SearchParameters parameters,
+            string text,
+            DocumentSearchResult<SearchDocument.Full> result,
+            TimeSpan duration)
+        {
+            var results = result.Results;
+            result.Results = null;
+
+            List<string> data;
+            switch (request.Type)
+            {
+                case AutocompleteRequestType.PackageIds:
+                    data = results.Select(x => x.Document.PackageId).ToList();
+                    break;
+
+                case AutocompleteRequestType.PackageVersions:
+                    data = results.SelectMany(x => x.Document.Versions).ToList();
```

Comment:
nit: I would throw if there is more than one result and then should do `[0]`. `SelectMany` across multiple IDs will never be the desired behavior AFAIK.

Reply:
Will the data need to be ordered?

Reply:
@joelverhagen I'll check that the results object has only one document as expected. `SelectMany` handles the case when there's no results succinctly, so I'll keep that as it is.

@cristinamanum The data does not need to be ordered, as per [the documentation](https://docs.microsoft.com/en-us/nuget/api/search-autocomplete-service-resource#response-1). However, I believe that versions will be ordered due to implementation details.

Reply:
An exception is thrown now if the results are malformed